List annex B features as optional on non-browsers

webbedspace opened the pull request:
Related to the discussion in #178, which asserts that the red colour for "No" entries in the Traceur/EJS/etc. columns is misleading for Annex B features.... here's a proposal that changes Annex B rows so that the results for non-browsers are de-emphasised, signifying that they are optional. The cells are drawn in gray, and a title attribute states that "This feature is optional on non-browser platforms."

![](http://i.imgur.com/DKfsZBi.png)

They still say "Yes" and "No", purely to allow the data to remain, but it isn't expressed as important to ES6 compatibility.

webbedspace committed:
Annex B features now listed as n/a in non-browsers

Since Annex B features are only required for browsers, non-browser
implementors now show a gray "n/a" instead of "No" when they're not
supported.

This also updates the feature-sorter code to account for non-applicable
features.

webbedspace committed:
Changed listing from "n/a" to grayed "Yes"/"No"

Now, both the presence or absence of an optional feature is drawn gray,
but still shows "Yes" or "No" anyway. Thus, the information regarding
implementation status is retained, but is highly de-emphasised.

Also, gray result squares now have an informative tooltip.

kangax commented:
That's not a bad idea! I was thinking to position those tests on the bottom of a table; they are from Annex after all. Perhaps also exclude from features counting. We can do both — position at the bottom and remove colors. What do you think?

webbedspace commented:
Actually, I forgot to mention: they already are excluded from the feature count for non-browsers (as in, EJS is "67/96" instead of "67/100", and is actually sorted higher than FF34 at 68/100).

kangax commented:
Ok good, so let's just put them on a bottom then (plus whatever you already changed)

webbedspace committed:
Shift annex B features to the bottom

To save on maintenance (and to produce a cleaner diff), this is done
programmatically.

webbedspace commented:
This is ready to go.

kangax commented on data-es6.js:
```diff
@@ -5300,3 +5310,7 @@ exports.tests = [
 ];
+
+//Shift annex B features to the bottom
+exports.tests = exports.tests.filter(function(e) { return !e.annex_b })
```
Won't work in pre-ES6 filter-less environments, but not sure we care...

webbedspace replied:
Well, the build script runs in Node, not the browser.

kangax replied:
Wasn't looking carefully. Never mind then :)
